### ISSUE TITLE
Options.extract does not handle missing port in URL

### ISSUE THREAD

**Issue by lcarva** [livereload/livereload-js#27]
When the livereload script tag does not provide a port in the URL, Options.extract falls back to 35729 which is incorrect. Instead, if no port is provided, it should use 80 for HTTP and 443 for HTTPS.

Example:
src="http://localhost/livreload.js" (Port is 80)
src="https://localhost/livreload.js" (Port is 443)

Maybe something like this:

``` javascript
if (mm[2]) {
  options.port = parseInt(mm[2], 10);
} else {
  options.port = options.https ? 443 : 80
}
```

**Comment by andreyvit** [comment]
Hey, thanks for raising this topic!

It's not universally true, though. Most servers _do_ listen on port 35729, including my very own GUI app.

Which server are you using that shares the port with the main web server?

**Comment by andreyvit** [comment]
Wait, I spoke too soon. :-) I see your point — if `livereload.js` is being loaded from port 80 (and not 35729, as it's supposed to), then maybe we should use that port to listening.

A change like that makes more sense, but will break workflow of some people — those who are using LR servers that don't come bundled with `livereload.js`.

That use case (servers that don't come bundled with `livereload.js`) is due to be deprecated, though.

**Comment by ocjojo** [comment]
#86 might be a fix.

With it you would be able to provide port 80 or 443 when embedding the script like so:
```html
<script src="http://localhost:80/livereload.js"></script>
```

**Comment by smhg** [comment]
With the port 80 and 443 fix from @ocjojo (version 3.0.1) this can indeed be somewhat handled.
Closing this, as a default port change would likely break too much.
